# Variant-Calling-in-RNA-seq-data-NA12878-
Variant Calling in NA12878 was performed using the GATK best practices.

# Fastq files and reference genome
# Setting up a conda environment on Mac
# Installing common bioinformtic tools
# Workflow 
### Running fastp
- Input 
 >Command 
- Output
### Removing adapters using fastp
- Input 
 >Command 
- Output
### Trimming the reference genome
 - Input 
 >Command 
- Outputt
### Generating a index for the reference genome
- Input 
 >Command 
- Output
### Mapping the fastq files to the trimmed reference genome
 - Input 
 >Command 
- Output
 
 # Moving analysis to AWS
 ## Starting a AWS EC2 instance
 ## Setting up a conda environment on AWS
 ## Installing common bioinformtic tools
 ## Expanding the memory on disk
 ## Transferring files from local computer to AWS
 ## Transferring files from S3 bucket to AWS
 ## Stopping the instance 
 ## Creating a AMI of the instance
 ## Terminating the instance 
 ## Using the AMI to generate an instance 
 # Workflow
 ### Running fastp
 - Input 
 >Command 
- Output
 ### Removing adapters using fastp
 - Input 
 >Command 
- Output
 ### Trimming the reference genome
  - Input 
 >Command 
- Output
 ### Generating a index for the reference genome
  - Input 
 >Command 
- Output
 ### Mapping the fastq files to the trimmed reference genome
 - Input 
 >Command 
- Output
 ### Mark duplicates
 - Input 
 >Command 
- Output
 ### SplitNtrim
  - Input 
 >Command 
- Output
 ### Base Recalibration
 - Input 
 >Command 
- Output
 ### Apply BQSR
 - Input 
 >Command 
- Output
 
# Variant calling
 
 


